### PR TITLE
MULE-17851: Ensure persistent connections closed when http listener stopped

### PR DESCRIPTION
**Pull request by eze210**

### CODE REVIEW

File: src/test/java/org/mule/service/http/impl/service/server/grizzly/GrizzlyHttpServerTestCase.java [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+import java.util.concurrent.ExecutorService;
+
+import org.apache.http.HttpVersion;
+import org.hamcrest.Matchers;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class GrizzlyHttpServerTestCase extends AbstractMuleContextTestCase {
```
Comment (afelisatti) [inline src/test/java/org/mule/service/http/impl/service/server/grizzly/GrizzlyHttpServerTestCase.java]: Can't you leverage the HttpServerTestCase?

File: src/test/java/org/mule/service/http/impl/service/server/grizzly/GrizzlyHttpServerTestCase.java [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+        writer.append(line).append("\r\n");
+      }
+      return writer.toString();
+    } catch (IOException e) {
+      return null;
+    }
+  }
+
+  private void assertResponse(String response, boolean shouldBeValid) {
+    assertThat(isEmpty(response), Matchers.is(!shouldBeValid));
```
Comment (afelisatti) [inline src/test/java/org/mule/service/http/impl/service/server/grizzly/GrizzlyHttpServerTestCase.java]: static import if possible

**Comment by eze210** [comment]
--validate

**Comment by eze210** [comment]
--validate

**Pull request merged**